### PR TITLE
Add meishi2 keyboard

### PR DESCRIPTION
**Pull request by Biacco42**
Add new keyboard "meishi2".

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/meishi2/meishi2.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
+    k00, k01, \
```
Comment (noroadsleft) [inline keyboards/meishi2/meishi2.h]:
```suggestion
    k00, k01, k10, k11  \
```

File: keyboards/meishi2/meishi2.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    k10, k11   \
```
Comment (noroadsleft) [inline keyboards/meishi2/meishi2.h]:
```suggestion
```

File: keyboards/meishi2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
+    KC_A, KC_B, \
```
Comment (noroadsleft) [inline keyboards/meishi2/keymaps/default/keymap.c]:
```suggestion
    KC_A, KC_B, KC_C, KC_D  \
```

File: keyboards/meishi2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    KC_C, KC_D \
```
Comment (noroadsleft) [inline keyboards/meishi2/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/meishi2/rules.mk [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/meishi2/rules.mk]:
```suggestion
```

File: keyboards/meishi2/keymaps/default/keymap.c (hunk 3 above)
Comment (fauxpark) [inline keyboards/meishi2/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater and less "hack-y".

**Comment by yanfali** [comment]
@Biacco42 you getting emails?

**Comment by Biacco42** [comment]
> Given the size, you could have used "direct pins" for the matrix.
> 
> https://docs.qmk.fm/#/hardware_avr?id=direct-pin-matrix

Thanks for your info. meishi2 is aiming for teaching materials for beginners, however, and so it uses matrix for their better keyboard mechanism understanding.

**Comment by Biacco42** [comment]
I am sorry to have kept you waiting. I finally fixed!

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**